metals/src/main/scala/scala/meta/internal/metals/CodeActionProvider.scala

```diff
@@ -46,4 +50,14 @@ final class CodeActionProvider(
     Future.sequence(actions).map(_.flatten)
   }
 
+  def executeCommands(
+      codeActionCommandData: CodeActionCommandData,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Option[CodeActionCommandResult] = {
+    codeActionCommandData match {
+      case data: ExtractMemberDefinitionData =>
+        Option(extractMemberAction.executeCommand(data))
+      case _ => None
```

Comment:
```suggestion
      case data => Future.failed(new CodeActionCommandNotFound(data.title))
```

we can create an additional exception. This should never happen, but if ever we do something wrong we can easily figure out what's happening instead of having it just skipped.